src/main/java/de/butbetter/keksys/VATCalculator.java

```diff
@@ -0,0 +1,29 @@
+package de.butbetter.keksys;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+@Service
+class VATCalculator implements Calculator {
+
+	private static final Logger log = LoggerFactory.getLogger(VATCalculator.class);
+
+	@Override
+	public Float calculateVAT(float price, float percent) {
+		float return_value = ((price / 100) * percent) + price;
+		log.info(price + " + " + percent + " = " + return_value);
+		return roundTo2DecimalPoints(return_value);
+	}
+
+	/**
+	 * rounds the value up to a given decimal point
+	 */
+	private float roundTo2DecimalPoints(float value) {
+		return new BigDecimal(value).setScale(2, RoundingMode.HALF_UP).floatValue();
```

Comment:
You cannot store infinity in a BigDecimal, 
this throws a NumberFormatException

Reply:
Issue detected: https://app.codacy.com/gh/But-Better/CalcApi/pullRequest?prid=8370557

Reply:
VATcalc now returns infinity like before

Reply:
wouldn't use this kind of check, since this is partially a code copy of the validator,
I'd rather try and catch the exception thrown and turn it into a ArithmeticException instead, which should be handled by either the Validator or the Controller at some point already

Reply:
or we catch the FormatException, and return the value anyways
all 3 could work

Reply:
decided to catch exception and just give back value when no rounding can be applied